velero-plugin-for-aws/config.go

```diff
@@ -29,27 +32,18 @@ func newAWSConfig(region, profile, credentialsFile string, insecureSkipTLSVerify
 		}
 		tr.TLSClientConfig.InsecureSkipVerify = insecureSkipTLSVerify
 	})
+	// Bucket: team-saas-backups-eu-west-1. Secret has same name because set by operator.
+	bucketCredentials, err := loadCredentialsFromSecret(bucket)
+	if err != nil {
+		return empty, errors.Wrapf(err, "could not load credentials from secret %s", bucket)
+	}
 	opts := []func(*config.LoadOptions) error{
 		config.WithRegion(region),
 		config.WithSharedConfigProfile(profile),
 		config.WithHTTPClient(client),
-	}
```

Comment:
can we not just set the AWS_SHARED_CREDENTIALS_FILE, this should also load credentials from a file, and you can just mount a secret?

Reply:
We can set the location of the license file, but there does not seem to be a way to guarantee that the file is used. I can set the file, but the SDK will still use the entire chain, find an IRSA role and not even check the file